cdap-unit-test/src/test/java/co/cask/cdap/test/app/TestFrameworkTestRun.java

```diff
@@ -554,6 +554,19 @@ public void testAppWithWorker() throws Exception {
     Assert.assertEquals(AppWithWorker.STOP, Bytes.toString(table.read(AppWithWorker.STOP)));
   }
 
+  @Test
+  public void testWorkerStop() throws Exception {
+    // Test to make sure the worker program's status goes to stopped after the run method finishes
+    ApplicationManager manager = deployApplication(NoOpWorkerApp.class);
+    WorkerManager workerManager = manager.getWorkerManager("DummyWorker");
+    try {
+      workerManager.start();
+      workerManager.waitForStatus(false, 5, 1);
+    } finally {
+      manager.stopAll();
```

Comment:
we don't need this manager.stopAll, right? nothing should be running at that point in time?